src/main/java/io/mosip/mimoto/controller/PresentationController.java

```diff
@@ -26,31 +26,42 @@ public class PresentationController {
     PresentationService presentationService;
     private final Logger logger = LoggerFactory.getLogger(PresentationController.class);
 
-    @Value("${mosip.inji.ovp.error.redirect.url}")
-    String injiOvpErrorRedirectUrl;
+    @Value("${mosip.inji.ovp.error.redirect.url.pattern}")
+    String injiOvpErrorRedirectUrlPattern;
 
     @Value("${mosip.inji.web.redirect.url}")
     String injiWebRedirectUrl;
 
     @GetMapping("/authorize")
-    public void performAuthorization(HttpServletResponse response, @ModelAttribute PresentationRequestDTO presentationRequestDTO) throws IOException {
+    public void performAuthorization(HttpServletResponse response,
+                                     @RequestParam("response_type") String responseType,
+                                     @RequestParam("resource") String resource,
+                                     @RequestParam("presentation_definition") String presentationDefinition,
+                                     @RequestParam("client_id") String clientId,
+                                     @RequestParam("redirect_uri") String redirectUri ) throws IOException {
         try {
             logger.info("Started Presentation Authorization in the controller.");
+            PresentationRequestDTO presentationRequestDTO = PresentationRequestDTO.builder()
+                    .responseType(responseType)
+                    .resource(resource)
+                    .presentationDefinition(presentationDefinition)
+                    .clientId(clientId)
+                    .redirectUri(redirectUri).build();
             String redirectString = presentationService.authorizePresentation(presentationRequestDTO);
             logger.info("Completed Presentation Authorization in the controller.");
             response.sendRedirect(redirectString);
         } catch( InvalidVerifierException exception){
             sendRedirect(response, injiWebRedirectUrl, exception.getErrorCode(), exception.getErrorText());
         } catch(VPNotCreatedException | InvalidCredentialResourceException exception){
-            sendRedirect(response, presentationRequestDTO.getRedirect_uri(), exception.getErrorCode(), exception.getErrorText());
+            sendRedirect(response, redirectUri, exception.getErrorCode(), exception.getErrorText());
         } catch (Exception exception){
-            sendRedirect(response, presentationRequestDTO.getRedirect_uri(), ErrorConstants.INTERNAL_SERVER_ERROR.getErrorCode(), ErrorConstants.INTERNAL_SERVER_ERROR.getErrorMessage());
+            sendRedirect(response, redirectUri, ErrorConstants.INTERNAL_SERVER_ERROR.getErrorCode(), ErrorConstants.INTERNAL_SERVER_ERROR.getErrorMessage());
```

Comment:
this redirect url again needs to be validated as per the config even before sending  the redirect.

Reply:
We are doing the verifiers redirect url verification in the verifierServiceImpl

Reply:
What happens when the exception is thrown before the verifierServiceImpl code is executed?

Reply:
Modified